crawler/twitter.py

```diff
@@ -0,0 +1,183 @@
+                {"userName": user["userName"],
+                "accessToken": user["accessToken"],
+                "accessTokenSecret": user["accessTokenSecret"]}
+            )
+
+        return twitterCredentials
+
+    def __get_twitter_users_queue(self):
+        usersQueue = []
+        results = self.db.run("MATCH (n:UsersQueue) RETURN n.id as id")
+        for user in results:
+            usersQueue.append({"id": user["id"]})
+
+        return usersQueue
+
+    def __validate_twitter_user(self, twitterHelper, userId):
+        api = TwitterAPI(
+            self.token,
+            self.secret,
+            twitterHelper["accessToken"],
+            twitterHelper["accessTokenSecret"])
+
+        response = api.request(self.USERS_SHOW, {"user_id": userId})
+        parsedResponse = json.loads(response.text)
+        isValid = False
+        if "status" in parsedResponse:
+            isValid = self.__is_language_valid(parsedResponse["status"]["lang"])
+        elif "errors" in parsedResponse:
+            isValid = self.__check_twitter_error_code(
+                parsedResponse["errors"][0]["code"]
+            )
+
+        return isValid
+
+    def __is_language_valid(self, language):
+        return {
+            self.GERMAN: True
+        }.get(language, False)
+
+    def __check_twitter_error_code(self, code):
+        return {
+            32: "Cannot authenticate",
+            50: "User not found",
+            63: "Suspended user",
+            88: "Switch helper",
+            89: "Switch helper",
+            131: "Internal error"
+        }.get(code, "Unknown error")
+
+    def __delete_twitter_user_from_queue(self, id):
+        query = "MATCH (n:UsersQueue {id: '" + id + "'}) "
+        query += "DETACH DELETE n"
+        response = self.db.run(query)
+        print("Twitter user with id: ", id, " deleted from users queue.")
+
+    def __create_users_queue_batch_query(self, key, followerId, followersCount):
+        if key == 0 and followersCount == 0:
+            key = "singleFollower"
+
+        query = "(:UsersQueue {id: '" + followerId + "'})"
+
+        return {
+            0: "CREATE " + query + ", ",
+            followersCount: query,
+            "singleFollower": "CREATE " + query
+        }.get(key, query + ", ")
+
+    def __create_user_followers_query(self, userId, followerId):
+        query = "MERGE (u:TwitterUser{id:'" + userId + "'}) "
+        query += "MERGE (f:TwitterUser{id:'" + followerId + "'}) "
+        query += "MERGE (u)<-[:FOLLOWS]-(f)"
+
+        return query
+
+    def __create_user_followers_batch_query(
+        self, key, userId, followerId, followersCount):
+
+        if key == 0 and followersCount == 0:
+            key = "singleFollower"
+
+        query = "MERGE (u"+str(key)+":TwitterUser{id:'" + userId + "'}) "
+        query += "MERGE (f"+str(key)+":TwitterUser{id:'" + followerId + "'}) "
+        query += "MERGE (u"+str(key)+")<-[:FOLLOWS]-(f"+str(key)+")"
```

Comment:
This is actually very important. First because the label :USER is already in the database, as well as the property 'uid', second, id is a built-in property of every kind of node in a neo4j database, a reserved word. If you use it, you will get the wrong results!